core/dbt/dataclass_schema.py

```diff
@@ -73,7 +73,9 @@ def __post_serialize__(self, dct):
     # performing the conversion to a dict
     @classmethod
     def __pre_deserialize__(cls, data):
-        if cls._hyphenated:
+        # `data` might not be a dict, e.g. for `query_comment`, which accepts
+        # a dict or a string; only snake-case for dict values.
+        if cls._hyphenated and isinstance(data, dict):
```

Comment:
Does this make sense @gshank?

Reply:
Looking at all the existing calls to this class method, it seems to be true.

Reply:
Honestly, I'm not sure exactly how this code gets reached, but I dropped into a debugger at this point on the unit test suite and found non-dictionary values--specifically, there's a test that sets `query-comment` to `""`. You should be able to reproduce the issue by dropping the `isinstance` check that I added and running something like `pytest -s test/unit/test_query_headers.py -k test_disable_query_comment`.

Reply:
You can also run that test with the following added to __pre_deserialize__:

```
        if cls._hyphenated and not isinstance(data, dict):
            import ipdb; ipdb.set_trace()
```

Looking at the stack, I'm inside a function called `__unpack_union_Project_query_comment__9f5a8d0e89384cc286d2b99acef5623d` that I'm guessing is dynamically generated and eval-ed by mashumaro 😱.